### PR TITLE
Issue #KN-50: Angular migration from version 9 to 10

### PR DESCRIPTION
**Pull request by gouravmore**
Issue #KN-50: Angular migration from version 9 to 10

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/pdf-player-wc/src/app/app.module.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { SunbirdPdfPlayerComponent } from '../../../sunbird-pdf-player/src/lib/s
     SunbirdPlayerSdkModule
   ],
   providers: [],
-  entryComponents: [SunbirdPdfPlayerComponent]
+  // entryComponents: [SunbirdPdfPlayerComponent]
```
Comment (harish-ennoventure) [inline projects/pdf-player-wc/src/app/app.module.ts:24]: Why this is commented

File: projects/sunbird-pdf-player/src/lib/services/viewer.service.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export class ViewerService {
   public showDownloadPopup: boolean;
   public src: string;
   public userName: string;
-  private metaData: any;
+  public metaData: any;
```
Comment (harish-ennoventure) [inline projects/sunbird-pdf-player/src/lib/services/viewer.service.ts:27]: why it is changed to public ?

File: projects/sunbird-pdf-player/src/lib/sunbird-pdf-player.service.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@ import { UtilService } from './services/util.service';
 export class SunbirdPdfPlayerService {
 
   private contentSessionId: string;
-  private playSessionId: string;
-  private telemetryObject: any;
+  public playSessionId: string;
```
Comment (harish-ennoventure) [inline projects/sunbird-pdf-player/src/lib/sunbird-pdf-player.service.ts:11]: why these are changed to public

**Review by harish-ennoventure — CHANGES_REQUESTED** [review]

File: projects/pdf-player-wc/src/app/app.module.ts (hunk 1 above)
Comment (gouravmore) [reply projects/pdf-player-wc/src/app/app.module.ts]: entryComponents is longer required with the Ivy compiler and runtime.

File: projects/sunbird-pdf-player/src/lib/services/viewer.service.ts (hunk 2 above)
Comment (gouravmore) [reply projects/sunbird-pdf-player/src/lib/services/viewer.service.ts]: Fixed and updated.

File: projects/sunbird-pdf-player/src/lib/sunbird-pdf-player.service.ts (hunk 3 above)
Comment (gouravmore) [reply projects/sunbird-pdf-player/src/lib/sunbird-pdf-player.service.ts]: Fixed and updated PR

File: projects/pdf-player-wc/src/app/app.module.ts [hunk 4]
```diff
@@ -20,8 +20,7 @@ import { SunbirdPdfPlayerComponent } from '../../../sunbird-pdf-player/src/lib/s
     HttpClientModule,
     SunbirdPlayerSdkModule
   ],
-  providers: [],
-  entryComponents: [SunbirdPdfPlayerComponent]
+  providers: []
```
Comment (HarishGangula) [inline projects/pdf-player-wc/src/app/app.module.ts:23]: Why entry component is removed

**Review by HarishGangula — CHANGES_REQUESTED** [review]
Have tested the player with all the scenarios and edge cases

File: projects/pdf-player-wc/src/app/app.module.ts (hunk 4 above)
Comment (gouravmore) [reply projects/pdf-player-wc/src/app/app.module.ts]: While upgrding the version from 9 to 10, 
they mentioned:=> ```In your application projects, you can remove entryComponents NgModules and any uses of ANALYZE_FOR_ENTRY_COMPONENTS. They are no longer required with the Ivy compiler and runtime.```
 we can remove it, if we want. For now, i keep this as it is and tested both scenarios.

**Pull request merged**